src/cljs/athens/keybindings.cljs

```diff
@@ -0,0 +1,231 @@
+     (.bind mousetrap-instance
+            keys
+            (fn [event]
+              (when stop-propagation? (.stopPropagation event))
+              (callback event)))
+     #(.unbind mousetrap-instance keys))))
+
+
+;; Same as mousetrap-bind but binds a map of keybindings (aliases-or-hotkeys and event handler pairs).
+;; API is a bit different since it accepts the mousetrap-instance as the first argument.
+;; and always stop the event propagation.
+(defn mousetrap-bind-all
+  [mousetrap-instance keybindings]
+  (let [unbind-fns
+        (doall
+          (map
+            (fn [[alias-or-hotkey callback-or-config]]
+              (let [callback (if (map? callback-or-config)
+                               (:callback callback-or-config)
+                               callback-or-config)
+                    stop-propagation? (if (map? callback-or-config)
+                                        (:stop-propagation? callback-or-config)
+                                        true)]
+                (mousetrap-bind
+                  alias-or-hotkey
+                  callback
+                  :stop-propagation? stop-propagation?
+                  :mousetrap-instance mousetrap-instance)))
+            keybindings))]
+    #(doall (map apply unbind-fns))))
+
+
+;; React component that listen binds event handlers to a local DOM element.
+;; It also listen for changes on keymap to re-bind the event handlers.
+;; Note: this is not meant to be used directly, use mousetrap instead.
+(defn mousetrap-binder
+  [_keymap keybindings _child]
+  (let [mousetrap-instance (r/atom nil)
+        unbind-all (r/atom #())
+        bind (fn [keybindings]
+               (reset! unbind-all
+                       (mousetrap-bind-all @mousetrap-instance keybindings)))]
+
+    (r/create-class
+      {:display-name
+       "mousetrap-binder"
+
+       :component-did-mount
+       #(bind keybindings)
+
+       :component-will-unmount
+       #(@unbind-all)
+
+       :component-did-update
+       (fn [this old-argv]
+         (let [new-keymap (second (r/argv this))
+               prev-keymap (second old-argv)
+               keybindings (nth (r/argv this) 2)]
+           (when (not= new-keymap prev-keymap)
+             (@unbind-all)
+             (bind keybindings))))
+
+       :reagent-render
+       (fn [_ _ child]
+         [:span
+          {:style {:display "contents"}
+           :ref #(reset! mousetrap-instance (new Mousetrap %))}
+          child])})))
+
+
+;; Wrap what is passed as "child" into a DOM node that listen for hotkeys that
+;; are bound to event handlers.
+;; If the application keymap changes, this component will re-bind the event
+;; handlers with the new hotkeys.
+;; keybindings is a pair of aliases-or-hotkeys and event handler pairs
+;; e.g: {:athena/toggle some-function1 "mod+z" some-function2}
+(defn mousetrap
+  [keybindings child]
+  (let [keymap @(subscribe [:keymap])]
+    [mousetrap-binder keymap keybindings child]))
```

Comment:
Are these the js `shenanigans`?

Reply:
Maybe `shenanigans` was too harsh, but yes.